napari_aicssegmentation/_tests/view/workflow_steps_view_test.py

```diff
@@ -22,3 +25,16 @@ def test_add_progress_bar(self):
 
     def test_add_workflow_steps(self):
         pass
+
+    def test_handle_modal_input_close_keep(self):
+        # Arrange
+        channel = Channel(0, "Brightfield")
+        self._view._controller.model.selected_channel = channel
+
+        self._view.close_keep = self._view.confirmation_modal.addButton("Close workflow", QMessageBox.AcceptRole)
+
+        # Act
+        self._view._handle_modal_input(self._view.close_keep)
```

Comment:
Instead of calling the private event handler directly, see if you can instead trigger a click on the actual button, if you have access to it.
Something like this: 
`self._view.close_keep.clicked.emit()`
or
`self._view.close_keep.click()`
I haven't done this with a inputs inside a modal so not 100% positive this will work, but i've done this with regular buttons and the dropdowns. The idea is that we can programmatically emit the "click" (or whichever) event when testing instead of just calling the handler method - which is the closest thing to actually running the app and physically clicking on the widgets. Most importantly, this validates that the event handlers are properly hooked up, while calling the private handler methods does not.

Reply:
I agree this is not suboptimal for testing the modal input, since what I'm really testing here is the resetting of the channel selection and not the handling of user input.

I can't find an example of something like what you suggested in our tests, do you mean you've written a test like that for another project? What I'm seeing on the internet seems to indicate that we need a `pytestqt` library to simulate user inputs for something like this.

I might just move this test to another test file, to test just the reset method.

Reply:
I had to delete `_handle_modal_input()` anyway (because I rolled it in to the controller as `close_workflow()` per your previous comment) so I just moved the test to the controller tests.

Reply:
You can see examples of this right here https://github.com/AllenCell/napari-aicssegmentation/blob/main/napari_aicssegmentation/_tests/view/workflow_select_view_test.py#L107
No need to simulate user inputs for unit tests - just emit the Widget's event.

Reply:
I see, good to know!